qiskit/transpiler/target.py

```diff
@@ -271,11 +279,36 @@ def __init__(
         qubit_properties=None,  # pylint: disable=unused-argument
         concurrent_measurements=None,  # pylint: disable=unused-argument
     ):
-        # A nested mapping of gate name -> qargs -> properties
-        self._gate_map = {}
-        self._coupling_graph = None
-        self._instruction_durations = None
-        self._instruction_schedule_map = None
+        super().__init__()
+
+    def get_non_global_operation_names(self, strict_direction=False):
+        """Return the non-global operation names for the target
+
+        The non-global operations are those in the target which don't apply
+        on all qubits (for single qubit operations) or all multi-qubit qargs
+        (for multi-qubit operations).
+
+        Args:
+            strict_direction (bool): If set to ``True`` the multi-qubit
+                operations considered as non-global respect the strict
+                direction (or order of qubits in the qargs is significant). For
+                example, if ``cx`` is defined on ``(0, 1)`` and ``ecr`` is
+                defined over ``(1, 0)`` by default neither would be considered
+                non-global, but if ``strict_direction`` is set ``True`` both
+                ``cx`` and ``ecr`` would be returned.
+
+        Returns:
+            List[str]: A list of operation names for operations that aren't global in this target
+        """
+        if strict_direction:
+            if self._non_global_basis_strict is None:
+                self._non_global_basis_strict = super()._get_non_global_operation_names(
+                    strict_direction
+                )
+            return self._non_global_basis_strict
+        if self._non_global_basis is None:
+            self._non_global_basis = super()._get_non_global_operation_names(strict_direction)
+        return self._non_global_basis
```

Comment:
Do we need to cache this?  I'm concerned that we no longer have any cache invalidation (not sure if we did before tbh).

Reply:
Oh sorry, I misread the cache invalidation below as being in the initialiser of `FakeTarget`.  In that case I'm fine with it.